### ISSUE TITLE
questions on config.json

### ISSUE THREAD

**Issue by bill-orange** [n8henrie/fauxmo#24]
- **Operating system and version**:
Raspian 3.2.17
- **Python version**:
3.4.4
- **Fauxmo version**:
latest release

## My Issue

What edits to I need to make to the config file to get fauxmo running enough to appear on my Echo?
Testing unedited, here's what I get:

fauxmo -c /home/pi/Desktop/config.json -v
2017-03-04 13:02:43 fauxmo:18       INFO     Fauxmo version 0.3.6
Traceback (most recent call last):
  File "/usr/local/bin/fauxmo", line 11, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.4/dist-packages/fauxmo/cli.py", line 28, in cli
    main(config_path=args.config, verbosity=verbosity)
  File "/usr/local/lib/python3.4/dist-packages/fauxmo/fauxmo.py", line 50, in main
    config = json.load(config_file)
  File "/usr/lib/python3.4/json/__init__.py", line 265, in load
    return loads(fp.read(),
  File "/usr/lib/python3.4/codecs.py", line 313, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xa9 in position 1916: invalid start byte

## [WHYT](http://mattgemmell.com/what-have-you-tried)

Tried it twice  My local network happens to be the same as the 'sample' config.  192.168.0.xxx

---

Please make sure you've taken these steps before submitting a new issue:

- [ ] Include the Python and Fauxmo version in your issue
- [ ] Ensure you're running a supported version of Python
    - Requires Python >= 3.2 for Fauxmo < 0.3.0
    - Requires Python >= 3.4.4 for Fauxmo >= 0.3.0
- [ ] Run Fauxmo in debug mode (`-vvv`) and include relevant output
- [ ] Search the existing (and closed) issues

**Comment by bill-orange** [comment]
Never mind.  Answered my own question.  I had to delete HOMEASSISTANT from the config.

One related question: rather than "on_cmd": going to a web server, I would like to alter the code to turn on and off LEDS on a PI.  Any starting advice?

**Comment by n8henrie** [comment]
Sweet! OpenSprinkler seems like a cool project.

You shouldn't have to delete the `HOMEASSISTANT` part, that's weird -- just leaving it blank should be fine.

Seems like the Pi-GPIO-Server should work fine -- Fauxmo only cares [that the response code is 200 or 201](https://github.com/n8henrie/fauxmo/blob/39e46e20e083054729b1a5bf65ff6ca19d938f4a/src/fauxmo/handlers/rest.py#L63). The approach you describe (having a server on `localhost` process `on_cmd`) is what I've done in the past, and works well.

**Comment by bill-orange** [comment]
Great Thanks.  I bought a Pi Zero Wireless yesterday from Microcenter.  in a few day I will test it and we will see if it has enough horsepower to support this code.

In your reply yesterday, you mentioned that leaving  HOMEASSISTANT  blank should be enough.  What did you mean?  Leave the IP address blank or something else.  I can retest with the old config.

**Comment by euridium** [comment]
hi bill-orange;  do you have an example of your json config for reading and writing to the  Pi-GPIO-Server please? if you ended up not using this for directly driving the GPIOs then how did you do it!
full working config would be a great starting point.

many thanks.

**Comment by n8henrie** [comment]
Glad you had some success!

You need to set up your pins appropriately in your program, I don't think that sounds like an appropriate modification for Fauxmo (although if I ever figure out how to enable user plugins that may be a perfect use case). In most programs that interact with the GPIO there is a requirement to "set up" before you change the PIN; if needed this should be done in the initialization of whatever program you're using.

A different port number for each device -- keep in mind that each Fauxmo device is "pretending" to be a different Wemo switch.

**Comment by bill-orange** [comment]
Got it.  I just noticed you documentation.  Very nice.  It answered my questions and a few more I did not know to ask. 

I was able to setup the GPIO ports on the target application, as you suggest.

I re-tested config with HomeAssistant Enable set to false rather than deleting the section.  It still crashed  Fauxmo.  I also deleted the whole enable line and it still crashed.  Odd, but since I am not using HomeAssistant it is not a problem for me.  I think I can close this issue out since I am successfully up and running three fake Wemo lights.
